### PR TITLE
add 2 more local source e2e tests

### LINKED ISSUE #451: Add follow-up e2e tests for local source type
Add e2e tests that validate the behavior in case the local source type is invalid:

- The local source points to a non-existent configmap
- The source points to a configmap with invalid bundle data

### PR DESCRIPTION
**Pull request by akihikokuroda**
Signed-off-by: akihikokuroda <akuroda@us.ibm.com>

Closes #451

### CODE REVIEW

File: test/e2e/plain_provisioner_test.go [hunk 1]
```diff
@@ -829,6 +829,136 @@ var _ = Describe("plain provisioner bundle", func() {
...
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "combo-local-",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: plain.ProvisionerID,
+					Source: rukpakv1alpha1.BundleSource{
+						Type: rukpakv1alpha1.SourceTypeLocal,
+						Local: &rukpakv1alpha1.LocalSource{
+							ConfigMapRef: &rukpakv1alpha1.ConfigMapRef{
+								Name:      "Non-exist",
```
Comment (exdx) [inline test/e2e/plain_provisioner_test.go:850]: nit: object names should be lowercase

File: test/e2e/plain_provisioner_test.go [hunk 2]
```diff
@@ -829,6 +829,136 @@ var _ = Describe("plain provisioner bundle", func() {
...
+			By("waiting until the bundle is reporting Failing state")
+			Eventually(func() bool {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(bundle), bundle); err != nil {
+					return false
+				}
+				if bundle.Status.Phase != rukpakv1alpha1.PhaseFailing {
+					return false
+				}
+				for _, condition := range bundle.Status.Conditions {
+					if condition.Reason == rukpakv1alpha1.ReasonUnpackFailed && condition.Message == "source bundle content: could not find configmap rukpak-system/Non-exist on the cluster" {
```
Comment (exdx) [inline test/e2e/plain_provisioner_test.go:875]:
```suggestion
					if condition.Reason == rukpakv1alpha1.ReasonUnpackFailed && condition.Message == "source bundle content: could not find configmap rukpak-system/non-exist on the cluster" {
```

**Review by exdx — APPROVED** [review]

File: test/e2e/plain_provisioner_test.go [hunk 3]
```diff
@@ -829,6 +829,136 @@ var _ = Describe("plain provisioner bundle", func() {
...
+			}
+			err := c.Create(ctx, bundle)
+			Expect(err).To(BeNil())
+		})
+
+		AfterEach(func() {
+			err := c.Delete(ctx, bundle)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
+		})
+		It("checks the bundle's phase gets failing", func() {
```
Comment (timflannagan) [inline test/e2e/plain_provisioner_test.go:865]: nit: eventually results in a failing bundle state?

File: test/e2e/plain_provisioner_test.go [hunk 4]
```diff
@@ -829,6 +829,136 @@ var _ = Describe("plain provisioner bundle", func() {
...
+				},
+			}
+			err = c.Create(ctx, bundle)
+			Expect(err).To(BeNil())
+		})
+
+		AfterEach(func() {
+			err := c.Delete(ctx, bundle)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
+		})
```
Comment (timflannagan) [inline test/e2e/plain_provisioner_test.go:942]: Delete the ConfigMap too?

**Review by timflannagan — COMMENTED** [review]
Looks reasonable to me. It looks like Dan had a couple of nits, but I'd like to ensure we're deleting the ConfigMap, and refactoring to use the gomega matchers for asserting status conditions. Thoughts?

File: test/e2e/plain_provisioner_test.go (hunk 1 above)
Comment (akihikokuroda) [reply test/e2e/plain_provisioner_test.go]: I change it.  Thanks!

File: test/e2e/plain_provisioner_test.go (hunk 2 above)
Comment (akihikokuroda) [reply test/e2e/plain_provisioner_test.go]: I change it.  Thanks!

File: test/e2e/plain_provisioner_test.go (hunk 3 above)
Comment (akihikokuroda) [reply test/e2e/plain_provisioner_test.go]: I change it.  Thanks!

File: test/e2e/plain_provisioner_test.go (hunk 4 above)
Comment (akihikokuroda) [reply test/e2e/plain_provisioner_test.go]: I change it.  Thanks!

**Comment by exdx** [comment]
Should be fine to merge since all comments were addressed

**Pull request merged**